packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1480,12 +1486,19 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
     if (_snackBars.isNotEmpty) {
       final ModalRoute<dynamic> route = ModalRoute.of(context);
       if (route == null || route.isCurrent) {
-        if (_snackBarController.isCompleted && _snackBarTimer == null)
-          _snackBarTimer = new Timer(_snackBars.first._widget.duration, () {
+        if (_snackBarController.isCompleted && _snackBarTimer == null) {
+          final SnackBar snackBar = _snackBars.first._widget;
+          _snackBarTimer = new Timer.periodic(snackBar.duration, (Timer timer) {
```

Comment:
When the user turns off accessibility while the snackbar is showing we will get some inconsistent behavior - the time the snackbar will remain visible will be somewhat random between 0 to snackBar.duration.

What do you think about listening to the accessibility change again and then firing a timer?

Reply:
It's not inconsistent, it's just undefined in the snackbar spec.

It's not a common use case to toggle TalkBack on and off while showing snackbars. This simplifies the implementation so we don't have to add/remove listeners on window.

Reply:
I think that from the user's perspective it's inconsistent: you turn off accessibility and the snack bar might disappear immediately or after a few seconds.

But I agree it's not a very important edge case so ok to avoid complicating the code for that.

Reply:
i don't think a periodic timer is a good idea here. we'd be waking the CPU regularly while accessibility is enabled, even though nothing is happening.

Reply:
please at a very minimum leave a TODO here saying that this will hurt battery for accessibility users.

Reply:
I've update this to use the didChangeDependencies lifecycle to close the snackbar if a11y is disabled - https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/scaffold.dart#L1407.

Shouldn't have a significant effect on the battery?